src/wp-includes/block-template-utils.php

```diff
@@ -473,6 +474,26 @@ function _remove_theme_attribute_in_block_template_content( $template_content )
 			unset( $blocks[ $key ]['attrs']['theme'] );
 			$has_updated_content = true;
 		}
+		if ( 'core/navigation' === $block['blockName'] && isset( $block['attrs']['ref'] ) ) {
+			unset( $blocks[ $key ]['attrs']['ref'] );
+			$has_updated_content = true;
+		}
+		if ( 'core/query' === $block['blockName'] && isset( $block['attrs']['queryId'] ) ) {
+			unset( $blocks[ $key ]['attrs']['queryId'] );
+			$has_updated_content = true;
+		}
```

Comment:
Are you sure `queryId` is a site-specific attribute? WordPress/gutenberg#42730 talks about removing ID's from the query block but I think it may have been referring to other attributes.

Reply:
@ntsekouras would you be able to weigh in here? 

This PR is proposing to strip all query blocks of the `queryId` attribute if the theme is exported using the site editor. My understanding is that the attribute is site specific and shouldn't be set explicitly in theme templates.

Reply:
Sorry for the late reply here. `queryId` is used for proper Query Pagination when we have multiple Query Loop blocks, and there are a few nuances here..

If `queryId` doesn't exist and a Query Loop block is inserterd in an editor, it will auto-generate one. We usually want to strip/omit this in block patterns, because we know it will be opened in editor and therefore created during insertion of the block. 

In themes now, if a template has more than one Query Loop blocks and we strip them to both, the pagination will be broken for these blocks. So a better strategy might be to preserve them.

Reply:
I see, thanks for the feedback, I agree in that case it's a better approach to leave the queryIds in. Updated in [642879b](https://github.com/WordPress/wordpress-develop/pull/3899/commits/642879bea57a5ac788610752627163bc4c049364).

Reply:
@carlomanf @ntsekouras How do you feel about unsetting `taxQuery` attribute on query block? 
A query block with a tax query `"taxQuery":{"category":[7]}` might give empty results in the new site.

Do you see any issues removing it or setting it to null?

Reply:
This makes sense. Similar props would be the `parents, author` and `exclude`(even if this one is not exposed in the UI.